File: 01_PictureMaker/python/screen.py
def new_screen(w, h):
    black = [0, 0, 0]
    result = []
    for x in range(w):
        result = result + [[]]
        for y in range(h):
            result[x] = result[x] + [black[:]]
    return result

def save_screen(screen, filename):
    FILE = open(filename, "w")
    lines = ["P3"]
    lines.append("" + str(len(screen)) + " " + str(len(screen[0])) + " 256")
    for x in range(len(screen)):
        next = ""
        for y in range(len(screen[x])):
            next = next + str(screen[x][y][0]) + " "
            next = next + str(screen[x][y][1]) + " "
            next = next + str(screen[x][y][2]) + " "
        lines.append(next)
    for i in range(len(lines)):
        lines[i] = lines[i] + "\n"
    FILE.writelines(lines)
    FILE.close()

def draw_pixel(screen, x, y, c):
    if x < len(screen) and x >= 0 and y >= 0 and y < len(screen[x]):
        screen[y][x] = c

def get_pixel(screen, x, y):
    return screen[y][x]

File: 01_PictureMaker/java/screen.java
import java.awt.Color;
import java.lang.IllegalArgumentException;
import java.io.PrintWriter;
import java.io.FileWriter;
import java.io.IOException;

public class screen
{
    private Color[][] arr;//r, c format
    public static int defaultW = 600;
    public static int defaultH = 600;
    
    public screen(int w, int h)
    {
	arr = new Color[h][w];
	for(int r = 0; r < h; r++)
	{
	    for(int c = 0; c < w; c++)
	    {
		arr[r][c] = new Color(0, 0, 0);
	    }
	}
    }
    public screen()
    {
	this(defaultW, defaultH);
    }
    
    public Color getPixel(int x, int y)
    {
	if(x >= 0 && x < arr[0].length && y >= 0 && y < arr.length)
	{
	    return arr[y][x];
	}
	throw new IllegalArgumentException();
    }
    
    public void setPixel(int x, int y, Color c)
    {
	if(x >= 0 && x < arr[0].length && y >= 0 && y < arr.length)
	{
	    arr[y][x] = c;
	}
	else
	{
	    throw new IllegalArgumentException();
	}
    }
    
    public void saveScreen(String filename)
    {
	PrintWriter pw = null;
	try
	{
	    pw = new PrintWriter(new FileWriter(filename));
	    pw.print("P3 \n");
	    pw.print(arr.length);
	    pw.print(" ");
	    pw.print(arr[0].length);
	    pw.print(" 256 \n");
	    for(int r = 0; r < arr.length; r++)
	    {
		for(int c = 0; c < arr[0].length; c++)
		{
		    pw.print(arr[r][c].getRed());
		    pw.print(" ");
		    pw.print(arr[r][c].getGreen());
		    pw.print(" ");
		    pw.print(arr[r][c].getBlue());
		    pw.print(" ");
		}
		pw.print("\n");
	    }
	    pw.flush();
	}
	catch(IOException e)
	{ }
	finally
	{
	    if(pw != null)
	    {
		pw.close();
	    }
	}
    }
}